### PR TITLE
Add heart effect

### PR DESCRIPTION
**Pull request by CicadaCinema**
Type: enhancement

---

Just another particle effect in the same style as the others
Here's what this looks like with my monitor size: https://streamable.com/e93kty

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add heart effect ([\#6188](https://github.com/matrix-org/matrix-react-sdk/pull/6188)). Contributed by @CicadaCinema.

Preview: https://pr6188--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by ColonisationCaptain** [comment]
Signed-off-by: Arseny Uskov <arsenyuskov@gmail.com>

File: src/effects/hearts/index.ts
Comment (t3chguy) [inline src/effects/hearts/index.ts:137]: heart != snowflake

File: src/effects/hearts/index.ts
Comment (t3chguy) [inline src/effects/hearts/index.ts:186]: this is very noisy

**Comment by ColonisationCaptain** [comment]
whoops, those are fixed now
not sure why that check is failing though

**Comment by gsouquet** [comment]
> whoops, those are fixed now
> not sure why that check is failing though

The check is failing due to a change in our pipelines that do not seem to play nicely with forked repository. I have opened a PR and will ping you once it is merged so that you can get this branch up to date with `develop` and the checks will pass

Thank you for bearing with me

**Comment by ColonisationCaptain** [comment]
> > whoops, those are fixed now
> > not sure why that check is failing though
> 
> The check is failing due to a change in our pipelines that do not seem to play nicely with forked repository. I have opened a PR and will ping you once it is merged so that you can get this branch up to date with `develop` and the checks will pass
> 
> Thank you for bearing with me

sounds good!

**Comment by gsouquet** [comment]
The PR has been merged, if you get this branch up to date with what I just merged in `develop` it should all get back to normal 🎊

**Comment by ColonisationCaptain** [comment]
Looks good!

**Review by turt2live — APPROVED** [review]
<3

File: src/effects/index.ts
Comment (nadonomy) [inline src/effects/index.ts:94]: This looks great!

One nit to merge— we don't really see much use of 💝 in the wild. Please can you amend to use 💞 in the timeline emote/message? Thanks!

**Review by nadonomy — CHANGES_REQUESTED** [review]
This looks great! Added one nit/amendment before merging. Thanks for the effort.

File: src/effects/index.ts
Comment (CicadaCinema) [reply src/effects/index.ts]: Personally I think gift_heart is more fitting in this context, but I've amended it anyway.

**Comment by daniellekirkwood** [comment]
Can you confirm that the only trigger action for this chat effect is "/heart" ?

**Comment by CicadaCinema** [comment]
> Can you confirm that the only trigger action for this chat effect is "/heart" ?

The other trigger is one of the following emojis: 💝 💖 💞 💕
I chose them because they are not used commonly enough, I think, for the effect to be triggered unintentionally.

**Comment by daniellekirkwood** [comment]
Thanks for your response, @CicadaCinema 

Our worry is that chat effects will slip from delightful to annoying if we have too many triggers, or different effects in the product at once. To that end, would it be possible to reduce the number of emoji's that trigger this response to 1?

**Let's go with one emoji trigger, the 💝  (gift_heart) emoji, as your animation is a gift and this seems to be the most appropriate emoji trigger.**

**Comment by CicadaCinema** [comment]
> Thanks for your response, @CicadaCinema
> 
> Our worry is that chat effects will slip from delightful to annoying if we have too many triggers, or different effects in the product at once. To that end, would it be possible to reduce the number of emoji's that trigger this response to 1?
> 
> **Let's go with one emoji trigger, the gift_heart (gift_heart) emoji, as your animation is a gift and this seems to be the most appropriate emoji trigger.**

Done!

**Comment by daniellekirkwood** [comment]
Thank you! Then this looks great :) 

Note, I've written documentation on Chat Effects and their strategy that will soon be available. It's waiting on a [PR review](https://github.com/vector-im/element-meta/pull/191) 

Thanks for your contribution, I'm excited to start using this effect.

**Review by daniellekirkwood — APPROVED** [review]
Have reviewed this PR from a product perspective. We took the number of emoji triggers down to 1 and I have written a document for future Chat Effects in Element.

Thanks!

File: src/effects/hearts/index.ts
Comment (dbkr) [inline src/effects/hearts/index.ts:2]: Copyright should be yourself (plus anywhere you've taken code from)

File: src/effects/hearts/index.ts
Comment (dbkr) [inline src/effects/hearts/index.ts:137]: If we're adding more chat effects by copying the existing ones, I'd really rather factor out the common code than copy & paste it between files (this is the same comment from the snowfall code).

File: src/effects/hearts/index.ts
Comment (CicadaCinema) [reply src/effects/hearts/index.ts:2]: done!

File: src/effects/hearts/index.ts
Comment (CicadaCinema) [reply src/effects/hearts/index.ts:138]: Are you sure it would be a good idea to refactor all the existing effects as part of this PR? Other effects, most recently the rainfall effect in Nov 2021 ( https://github.com/matrix-org/matrix-react-sdk/pull/7086 ) have been merged without any issues raised about copied code.

Comment (turt2live) [reply src/effects/hearts/index.ts:138]: It's a long-term concern from before the rainfall effect as well. In general we aim for each PR to be better than the last, improving the situation as we go.

Given this effect is one of the last we'll be merging though, I'm not overly concerned with code duplication personally. The area could use work, but that can be a different PR. If it was included in this PR, all the better.

File: src/effects/index.ts
Comment (turt2live) [inline src/effects/index.ts:94]: For consistency with the trigger emoji

```suggestion
        fallbackMessage: () => _t("sends hearts") + " 💝",
```

**Review by turt2live — APPROVED** [review]
Thanks!

**Comment by turt2live** [comment]
hmm, looks like you've used your `develop` branch to contribute this - generally we don't recommend this as it now makes updating your fork a bit harder (but not impossible). 

I'm going to open a draft PR to ensure the CI is happy with the change and go from there.

**Comment by CicadaCinema** [comment]
> hmm, looks like you've used your `develop` branch to contribute this - generally we don't recommend this as it now makes updating your fork a bit harder (but not impossible).
> 
> I'm going to open a draft PR to ensure the CI is happy with the change and go from there.

Thank you very much! I realised this was a mistake some time ago but I thought making a new PR against a new branch might just inconvenience everyone involved here - didn't know that it could mess up the CI too, though.

**Comment by turt2live** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/8271 was happy, so force-merging

**Pull request merged**
